src/incomingDelivery/controller.js

```diff
@@ -101,5 +101,24 @@ function getQueue(req, res) {
  * 
  */
 function setReceived(req, res) {
+  ExpressHandler( 
+    async () => { 
+      const { _id } = req.body;
+      const userId = req.user._id
+      
+      const incomingDelivery = await IncomingDelivery.findOne({ _id });
+      if ( incomingDelivery.receivedOn ) return HTTPError('delivery already received');
+
+      incomingDelivery.receivedOn = new Date();
```

Comment:
This is actually fine.
`isDueBackOn` is meant to only capture a calendar date, so a string is actually preferred--to avoid timezone logic.
`receivedOn` is a time stamp, so we care about both date and time.